File: solutions/c/hamming/1/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs) {
  int distance = 0;

  for (int i = 0; ; i++) {
    char l = lhs[i];
    char r = rhs[i];

    if (l == 0)
      if (r == 0)
        break;
      else
        return -1;
    else
      if (r == 0)
        return -1;

    if (l != r)
      distance++;
  }

  return distance;
}

File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
    if string.len(a) ~= string.len(b) then
        return -1
    end

    local result = 0

    for i = 1, string.len(a) do
        if (a:byte(i) ~= b:byte(i)) then
            result = result + 1
        end
    end

    return result
end

return Hamming
